internal/provider/kubernetes/controller.go

```diff
@@ -434,6 +416,50 @@ func (r *gatewayAPIReconciler) processSecurityPolicySecretRefs(
 					"policy", policy, "secretRef", basicAuth.Users)
 			}
 		}
+
+		// Add the referenced BackendRefs and ReferenceGrants in ExtAuth to Maps for later processing
+		extAuth := policy.Spec.ExtAuth
+		if extAuth != nil {
+			var backendRef gwapiv1.BackendObjectReference
+			if extAuth.GRPC != nil {
+				backendRef = extAuth.GRPC.BackendRef
+			} else {
+				backendRef = extAuth.HTTP.BackendRef
+			}
+
+			backendNamespace := gatewayapi.NamespaceDerefOr(backendRef.Namespace, policy.Namespace)
```

Comment:
can you make sure that if a backendTLSPolicy exists for this backendRef that is included into `resources` as well ?

sidenote (not directly related to this PR) I dont see any tls context in the translation https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/gatewayapi/securitypolicy.go#L753
shouldnt we be reusing this function ?
https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/gatewayapi/route.go#L1074

Reply:
> shouldnt we be reusing this function ?

I have a TODO in the code and plan to do that but maybe later after v1.0.0. It requires massive modification on the current xRoute translation because the current Destination processing is tied to RouteParentContext.